app/models/comment.php

```diff
@@ -0,0 +1,46 @@
+<?php
+class Comment extends AppModel
+{
+	public $validation = array(
+		'body' => array(
+			'length' => array(
+				'validate_between', 1, 200,
+				),
+			),
+		);
+
+	public function getAll($offset, $limit)
+	{
+		$comments = array();
+		$thread = Param::get('thread_id');
+		$db = DB::conn();
+
+		$rows = $db->rows(
+		"SELECT * FROM comment WHERE thread_id = ? ORDER BY created ASC LIMIT {$offset},{$limit}", array($thread));
+
+		foreach ($rows as $row) {
+			$comments[] = new self($row);
+		}
+
+		return $comments;
+	}
+
+	public static function countAll()
+	{
+		$db = DB::conn();
+		$id = Param::get('thread_id');
+		return (int)$db->value("SELECT COUNT(*) FROM comment WHERE thread_id = {$id}");
```

Comment:
Already removed the casting po.
Link https://github.com/renzosunico/BoardExercises/commit/29cd94398e8f5d95e6a2834785064131154c8c14